Test case test_verify_storageclassclaim failed

The test case given below is failing in FaaS platform.
tests/managed-service/test_storageclassclaim.py::TestStorageClassClaim::test_verify_storageclassclaim

```
        log.info(f"Waiting for storageclassclaim {sc_claim_obj_rbd.name} to be Ready")
>       sc_claim_obj_rbd.ocp.wait_for_resource(
            condition=constants.STATUS_READY,
            resource_name=sc_claim_obj_rbd.name,
            column="PHASE",
            resource_count=1,
        )

tests/managed-service/test_storageclassclaim.py:47: 

......
......
>       return resource_info[column_index]
E       IndexError: list index out of range

ocs_ci/ocs/ocp.py:877: IndexError

```
Test report : https://ocs4-jenkins-csb-odf-qe.apps.ocp-c1.prod.psi.redhat.com/job/qe-odf-multicluster/1710/testReport/

--- a/ocs_ci/ocs/resources/storageclassclaim.py
+++ b/ocs_ci/ocs/resources/storageclassclaim.py
@@ -6,6 +6,7 @@ import logging
 
 from ocs_ci.framework import config
 from ocs_ci.helpers.helpers import create_unique_resource_name
+from ocs_ci.ocs.ocp import OCP
 from ocs_ci.ocs.resources.ocs import OCS
 from ocs_ci.ocs import constants
 from ocs_ci.utility import templating
@@ -52,6 +53,8 @@ def create_storageclassclaim(
     interface_type,
     storage_class_claim_name=None,
     namespace=None,
+    storageclient_name=None,
+    storageclient_namespace=None,
 ):
     """
     Create a storageclassclaim
@@ -83,14 +86,32 @@ def create_storageclassclaim(
             f"test-{interface_type.lower()}", constants.STORAGECLASSCLAIM.lower()
         )
     )
+
+    if config.ENV_DATA["platform"] == constants.FUSIONAAS_PLATFORM:
+        # Get the storageclient name and namespace if not available
+        if not (storageclient_name and storageclient_namespace):
+            storageclient_obj = OCP(
+                kind=constants.STORAGECLIENT,
+                namespace=storageclient_namespace
+                or config.ENV_DATA["cluster_namespace"],
+                resource_name=storageclient_name if storageclient_name else "",
+            )
+            storageclient_data = (
+                storageclient_obj.get(resource_name=storageclient_name)
+                if storageclient_name
+                else storageclient_obj.get()["items"][0]
+            )
+            storageclient_name = storageclient_data["metadata"]["name"]
+            storageclient_namespace = storageclient_data["metadata"]["namespace"]
+        sc_claim_data["spec"]["storageClient"] = {
+            "name": storageclient_name,
+            "namespace": storageclient_namespace,
+        }
+        # Storageclassclaim is a cluster scoped resource in ODF versions supported in FaaS
+        namespace = None
+
     if namespace:
-        if config.ENV_DATA["platform"] == constants.FUSIONAAS_PLATFORM:
-            sc_claim_data["spec"]["storageClient"] = {
-                "name": "storageclient",
-                "namespace": namespace,
-            }
-        else:
-            sc_claim_data["metadata"]["namespace"] = namespace
+        sc_claim_data["metadata"]["namespace"] = namespace
 
     sc_claim_obj = StorageClassClaim(**sc_claim_data)
     sc_claim_obj.create(do_reload=True)

--- a/ocs_ci/ocs/resources/storageclassclaim.py
+++ b/ocs_ci/ocs/resources/storageclassclaim.py
@@ -64,6 +64,8 @@ def create_storageclassclaim(
             (e.g. CephBlockPool, CephFileSystem)
         storage_class_claim_name (str): The name of storageclassclaim to create
         namespace(str): The namespace in which the storageclassclaim should be created
+        storageclient_name(str): Name of the storageclient
+        storageclient_namespace(str): Namespace of storageclient
 
     Returns:
         OCS: An OCS instance for the storageclassclaim

--- a/tests/managed-service/test_storageclassclaim.py
+++ b/tests/managed-service/test_storageclassclaim.py
@@ -8,6 +8,7 @@ from ocs_ci.framework.testlib import (
     ms_consumer_required,
     skipif_ocs_version,
 )
+from ocs_ci.ocs.exceptions import TimeoutExpiredError, ResourceWrongStatusException
 from ocs_ci.ocs.resources.storageclassclaim import create_storageclassclaim
 from ocs_ci.helpers.helpers import create_ocs_object_from_kind_and_name
 from ocs_ci.utility.utils import TimeoutSampler
@@ -49,12 +50,25 @@ class TestStorageClassClaim(ManageTest):
         )
 
         for sc_claim in [sc_claim_obj_rbd, sc_claim_obj_cephfs]:
-            for claim_info in TimeoutSampler(timeout=60, sleep=3, func=sc_claim.get):
-                if claim_info.get("status", {}).get("phase") == constants.STATUS_READY:
-                    log.info(
-                        f"Storageclassclaim {sc_claim.name} {constants.STATUS_READY}"
-                    )
-                    break
+            try:
+                for claim_info in TimeoutSampler(
+                    timeout=60, sleep=3, func=sc_claim.get
+                ):
+                    if (
+                        claim_info.get("status", {}).get("phase")
+                        == constants.STATUS_READY
+                    ):
+                        log.info(
+                            f"Storageclassclaim {sc_claim.name} {constants.STATUS_READY}"
+                        )
+                        break
+            except TimeoutExpiredError:
+                raise ResourceWrongStatusException(
+                    sc_claim,
+                    describe_out=sc_claim.describe(),
+                    column="PHASE",
+                    expected=constants.STATUS_READY,
+                )
 
         # Get OCS object of kind storageclass for both the storageclasses created by storageclassclaims
         sc_obj_rbd = create_ocs_object_from_kind_and_name(

--- a/tests/managed-service/test_storageclassclaim.py
+++ b/tests/managed-service/test_storageclassclaim.py
@@ -52,7 +52,7 @@ class TestStorageClassClaim(ManageTest):
         for sc_claim in [sc_claim_obj_rbd, sc_claim_obj_cephfs]:
             try:
                 for claim_info in TimeoutSampler(
-                    timeout=60, sleep=3, func=sc_claim.get
+                    timeout=500, sleep=10, func=sc_claim.get
                 ):
                     if (
                         claim_info.get("status", {}).get("phase")

--- a/tests/managed-service/test_storageclassclaim.py
+++ b/tests/managed-service/test_storageclassclaim.py
@@ -68,6 +68,7 @@ class TestStorageClassClaim(ManageTest):
                     describe_out=sc_claim.describe(),
                     column="PHASE",
                     expected=constants.STATUS_READY,
+                    got=sc_claim.get().get("status", {}).get("phase"),
                 )
 
         # Get OCS object of kind storageclass for both the storageclasses created by storageclassclaims

--- a/tests/managed-service/test_storageclassclaim.py
+++ b/tests/managed-service/test_storageclassclaim.py
@@ -52,7 +52,7 @@ class TestStorageClassClaim(ManageTest):
         for sc_claim in [sc_claim_obj_rbd, sc_claim_obj_cephfs]:
             try:
                 for claim_info in TimeoutSampler(
-                    timeout=60, sleep=3, func=sc_claim.get
+                    timeout=180, sleep=10, func=sc_claim.get
                 ):
                     if (
                         claim_info.get("status", {}).get("phase")

--- a/tests/managed-service/test_storageclassclaim.py
+++ b/tests/managed-service/test_storageclassclaim.py
@@ -52,7 +52,7 @@ class TestStorageClassClaim(ManageTest):
         for sc_claim in [sc_claim_obj_rbd, sc_claim_obj_cephfs]:
             try:
                 for claim_info in TimeoutSampler(
-                    timeout=500, sleep=10, func=sc_claim.get
+                    timeout=120, sleep=5, func=sc_claim.get
                 ):
                     if (
                         claim_info.get("status", {}).get("phase")

--- a/tests/managed-service/test_storageclassclaim.py
+++ b/tests/managed-service/test_storageclassclaim.py
@@ -52,7 +52,7 @@ class TestStorageClassClaim(ManageTest):
         for sc_claim in [sc_claim_obj_rbd, sc_claim_obj_cephfs]:
             try:
                 for claim_info in TimeoutSampler(
-                    timeout=180, sleep=10, func=sc_claim.get
+                    timeout=300, sleep=10, func=sc_claim.get
                 ):
                     if (
                         claim_info.get("status", {}).get("phase")

--- a/tests/managed-service/test_storageclassclaim.py
+++ b/tests/managed-service/test_storageclassclaim.py
@@ -49,7 +49,7 @@ class TestStorageClassClaim(ManageTest):
         )
 
         for sc_claim in [sc_claim_obj_rbd, sc_claim_obj_cephfs]:
-            for claim_info in TimeoutSampler(timeout=180, sleep=10, func=sc_claim.get):
+            for claim_info in TimeoutSampler(timeout=60, sleep=3, func=sc_claim.get):
                 if claim_info.get("status", {}).get("phase") == constants.STATUS_READY:
                     log.info(
                         f"Storageclassclaim {sc_claim.name} {constants.STATUS_READY}"

--- a/tests/managed-service/test_storageclassclaim.py
+++ b/tests/managed-service/test_storageclassclaim.py
@@ -10,6 +10,7 @@ from ocs_ci.framework.testlib import (
 )
 from ocs_ci.ocs.resources.storageclassclaim import create_storageclassclaim
 from ocs_ci.helpers.helpers import create_ocs_object_from_kind_and_name
+from ocs_ci.utility.utils import TimeoutSampler
 
 log = logging.getLogger(__name__)
 
@@ -43,23 +44,17 @@ class TestStorageClassClaim(ManageTest):
         teardown_factory(sc_claim_obj_rbd)
         teardown_factory(sc_claim_obj_cephfs)
 
-        log.info(f"Waiting for storageclassclaim {sc_claim_obj_rbd.name} to be Ready")
-        sc_claim_obj_rbd.ocp.wait_for_resource(
-            condition=constants.STATUS_READY,
-            resource_name=sc_claim_obj_rbd.name,
-            column="PHASE",
-            resource_count=1,
+        log.info(
+            f"Waiting for storageclassclaims {sc_claim_obj_rbd.name} and {sc_claim_obj_cephfs.name} to be Ready"
         )
 
-        log.info(
-            f"Waiting for storageclassclaim {sc_claim_obj_cephfs.name} to be Ready"
-        )
-        sc_claim_obj_cephfs.ocp.wait_for_resource(
-            condition=constants.STATUS_READY,
-            resource_name=sc_claim_obj_cephfs.name,
-            column="PHASE",
-            resource_count=1,
-        )
+        for sc_claim in [sc_claim_obj_rbd, sc_claim_obj_cephfs]:
+            for claim_info in TimeoutSampler(timeout=180, sleep=10, func=sc_claim.get):
+                if claim_info.get("status", {}).get("phase") == constants.STATUS_READY:
+                    log.info(
+                        f"Storageclassclaim {sc_claim.name} {constants.STATUS_READY}"
+                    )
+                    break
 
         # Get OCS object of kind storageclass for both the storageclasses created by storageclassclaims
         sc_obj_rbd = create_ocs_object_from_kind_and_name(

--- a/tests/managed-service/test_storageclassclaim.py
+++ b/tests/managed-service/test_storageclassclaim.py
@@ -52,7 +52,7 @@ class TestStorageClassClaim(ManageTest):
         for sc_claim in [sc_claim_obj_rbd, sc_claim_obj_cephfs]:
             try:
                 for claim_info in TimeoutSampler(
-                    timeout=300, sleep=10, func=sc_claim.get
+                    timeout=60, sleep=3, func=sc_claim.get
                 ):
                     if (
                         claim_info.get("status", {}).get("phase")

